lib/jsonapi/plugs/query_parser.ex

```diff
@@ -246,12 +246,29 @@ defmodule JSONAPI.QueryParser do
 
   @spec get_view_for_type(module(), String.t()) :: module() | no_return()
   def get_view_for_type(view, type) do
-    case Enum.find(view.relationships, fn {k, _v} -> Atom.to_string(k) == type end) do
+    case Enum.find(view.relationships, fn relationship ->
+           is_field_valid_for_relationship(relationship, type)
+         end) do
       {_, view} -> view
       nil -> raise_invalid_field_names(type, view.type())
     end
   end
 
+  @spec is_field_valid_for_relationship({atom(), module()}, String.t()) :: boolean()
```

Comment:
This was the simplest way I could think of to handle the deprecation, but happy to take suggestions if needed.